### PR TITLE
Cleanup CI workflows

### PR DESCRIPTION
**Pull request by gaby**
- Remove redundant comments.
- Fix go-versions inconsistency across ci files.
- Add '*' for branches in pull request.
- Add govulncheck workflow.
- Rename workflows with the tool name.
- Merge changes from #319 here

### CODE REVIEW

File: .github/workflows/sync-docs.yml [hunk 1]
```diff
@@ -27,9 +27,6 @@ jobs:
               with:
                   node-version: '18'
 
-            - name: Install JQ
-              run: sudo apt-get install jq
```
Comment (gaby) [inline .github/workflows/sync-docs.yml:31]: Runners have `jq` pre-installed.

**Comment by gaby** [comment]
@ReneWerner87 I noticed the tests didn't run in https://github.com/gofiber/template/pull/316 when i added the "*" rule for pull-requests branches it triggered the CI. Thus why I did this PR

**Comment by gaby** [comment]
@ReneWerner87 Should we add a rule that benchmarks only run if changes to `.go` files are detected?

**Comment by ReneWerner87** [comment]
Yes sounds good

**Review by ReneWerner87 — APPROVED** [review]

**Comment by gaby** [comment]
@efectn @ReneWerner87 Done, benchmark will only run on changes to `.go` or `go.mod` files.

**Pull request merged**